drivers: display: make ssd1306 driver i2c compatible

rbudai98 opened the pull request:
SSD1306 supports I2C as well, besides SPI

## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have complied with the [Submission Checklist](http://analogdevicesinc.github.io/no-OS/contributing.html#submission-checklist)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

amiclaus commented on drivers/display/ssd_1306/ssd_1306.c:
```diff
@@ -57,6 +58,25 @@ const struct display_controller_ops ssd1306_ops = {
+		buff_tmp[0] = 0x00;
+		memcpy(&buff_tmp[1], command, length);
+		ret = no_os_i2c_write(extra_param->i2c_desc, buff_tmp, length + 1, 1);
+	} else {
```
hmm, maybe this else can be dropped if ret is initialized from the start with an error code.

amiclaus commented on drivers/display/ssd_1306/ssd_1306.c:
```diff
@@ -235,11 +283,21 @@ int32_t ssd_1306_remove(struct display_dev *device)
+			return -1;
+	}
+
+	if (extra->comm_type == SSD1306_SPI)
```
switch case for improved readability.

amiclaus reviewed (commented):
small comments, otherwise lgtm.

rbudai98 commented:
v2:
* added switch cases
* data and command transmit is differentiated with the help of a GPIO pin in case of SPI communication and extra byte transfer in case of I2C. Moved this logic in a single transmit function.
